teammapper-backend/src/map/services/maps.service.ts

```diff
@@ -608,14 +630,21 @@ export class MapsService {
     }
   }
 
-  async createEmptyMap(rootNode?: IMmpClientNodeBasics): Promise<MmpMap> {
+  async createEmptyMap(
+    rootNode?: IMmpClientNodeBasics,
+    userId?: string
+  ): Promise<MmpMap> {
     const newMap: MmpMap = this.mapsRepository.create()
     const savedNewMap: MmpMap = await this.mapsRepository.save(newMap)
 
     if (rootNode) {
       await this.createRootNodeForMap(rootNode, savedNewMap.id)
     }
 
+    if (userId) {
+      await this.mapsRepository.update(savedNewMap.id, { owner: userId })
```

Comment:
why do we need to update? we can simply merge the user id in the save / create call?